### PR TITLE
Add support for Control4 Rooms

### PR DESCRIPTION
**Pull request by kdkavanagh**
Will be accompanied by HomeAssistant `media_player` support

### CODE REVIEW

**Comment by nalin29** [comment]
Additionally, I have noticed that this is not integrated with the websocket implementation it seems that the websocket cannot subscribe to room updates. Is there a way I can test and get my own websocket working through trial and error. What were the steps taken to capture api and websocket messages?

**Comment by kdkavanagh** [comment]
The websocket API seemed to require some unique handshake process to fully connect.  Beyond that, the two methods seemed functionally similar and I didnt need any time-sensitive push updates into HA.  It was a bit easier to just use the REST API, especially since the rest of this library was using that as well.  

To reverse engineer the API, I had to MITM the android app installed on an old android phone I had lying around - I needed to rebuild the apk with custom CA certs and use BurpSuite Community edition to intercept the API calls from the app.  Unfortunately I dont recall the exact set of steps, I just remember it was a bit of a pain, but there are tutorials out there since this is a pretty common need for app developers

**Comment by kdkavanagh** [comment]
+1 re: the merge question.  In the meantime, I've been using a local install of this plugin+the HA integration I wrote.  I can probably get those HA changes cleaned up and PR'd if that'd be helpful

**Comment by nalin29** [comment]
Thank you. Yeah I installed your library and was able to change to current HA plugin to support my media rooms for source and volume/power control using the current polling approach. It would be better to be push and not as hard coded.

File: pyControl4/room.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            {"deviceid": source_id},
+        )
+
+    async def setAudioSource(self, source_id: int):
+        """Sets the current audio source for the room"""
+        await self.setSource(source_id, audio_only=True)
+
+    async def setVideoAndAudioSource(self, source_id: int):
+        """Sets the current audio and video source for the room"""
+        await self.setSource(source_id, audio_only=False)
```
Comment (lawtancool) [inline pyControl4/room.py:48]: Since we have two different set audio/set audio and video methods, the `setSource` method should be private (`_setSource`)?

**Review by lawtancool — CHANGES_REQUESTED** [review]

**Comment by lawtancool** [comment]
Closed by #21

**Pull request closed without merging**